src/de/espend/idea/laravel/ui/ExtractPartialViewDialog.java

```diff
@@ -0,0 +1,75 @@
+package de.espend.idea.laravel.ui;
+
+import com.intellij.openapi.fileTypes.FileTypes;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.ui.ValidationInfo;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.ui.EditorTextField;
+import de.espend.idea.laravel.blade.actions.NewViewNameCompletionProvider;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.regex.Pattern;
+
+public class ExtractPartialViewDialog extends DialogWrapper {
+
+    private EditorTextField viewNameEditor;
+
+    private JPanel panel;
+
+    // View name should be "header", or "partials.header", or "sh-ar_ed.partials.header",
+    // so, words with '_' and '-' separated with dots
+    private static Pattern viewNamePattern = Pattern.compile("^[\\w\\d\\-_]+([\\.\\/][\\w\\d\\-_]+)*$");
```

Comment:
this there i way to support namespace also: `de.espend.idea.laravel.blade.util.BladeTemplateUtil#resolveTemplateName`?